### ISSUE TITLE
add option to hide tree or matrix

### ISSUE THREAD

**Issue by bmesuere** [unipept/unipept#306]
Add option to manually hide the tree or matrix.

Auto hide matrix if too many genomes are present:

> if all E. coli and Shigella genomes are clustered together, the tree labels become larger than the row height, so that the labels start to overlap horizontally; in this case, either the matrix should be hidden and the tree should be shown with a larger row height (that fits the labels) or the font size should be reduced
- auto hide matrix when loading many genomes
- auto hide tree when exporting unclustered matrix

_[Original issue](https://github.ugent.be/unipept/unipept/issues/306) by @bmesuere on Sun May 11 2014 at 14:29._

**Comment by pdawyndt** [comment]
we could also add an option to hide the labels at the bottom, at the right, or both


_[Original comment](https://github.ugent.be/unipept/unipept/issues/306#issuecomment-16132) by @pdawyndt on Fri Apr 17 2015 at 12:37._

**Comment by bmesuere** [comment]
I first implemented this by adding a global boolean `matrixHidden` that was used to set the `display` css property to hidden for the background, cells and bottom labels.
To make use of the extra horizontal space, the tree width must take into account the matrix status on construction.
This works ok-ish, but there's a bottom gap of 200px where the labels used to be. To also use this space, a second y-scale was added and used appropriately. The matrix and tree width and height functions were moved to separate functions. 
While these changes display the tree at full width with only the right labels showing, the user experience is suboptimal. Each time an element is added, moved or reordered, the tree gets blurred.
We could work around this by not blurring it when the matrix is hidden, but this is somewhat of a hack since there is extra code needed in at least 6 places. This would also break the optimisation of not calculating the similarity if the sim matrix tab is not shown.

My (broken) implementation already required 150+ changes in a file of 600 lines and further fixing edge cases, image exports and a user interface to show/hide the matrix would more than double this and would only result in a hacked-on feature.

To implement this the right way, we would need to rewrite the entire similarity matrix with this kind of modularity in mind. That why I suggest to cut this feature for now.


_[Original comment](https://github.ugent.be/unipept/unipept/issues/306#issuecomment-18642) by @bmesuere on Mon Jun 22 2015 at 18:27._
